From cd59400927a43947aa7effd8b6e2a31b1a163266 Mon Sep 17 00:00:00 2001
From: "Shi, Zhongjie" <zhongjie.shi@intel.com>
Date: Fri, 4 Aug 2017 07:53:15 +0000
Subject: [PATCH 4263/4301] Add SELinux class for SocketCAN.

commit f9a22a8ef4f3bd6849565a9ae5194605362b9680 from
git://git@github.com:01org/linux-intel-4.9.git

This was done independently without referencing others'
work although it looks the same as the commit in another
code base:
https://android.googlesource.com/kernel/msm/+/
28b6f2d6e9653ef252a4e0bd3211c56d7200b67e%5E%21/

Change-Id: I54c08ccce3a458f4ad866e01bfb5e98949ed2a4d
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-48496
Signed-off-by: Shi, Zhongjie <zhongjie.shi@intel.com>
---
 security/selinux/hooks.c            |    2 ++
 security/selinux/include/classmap.h |    2 ++
 2 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/security/selinux/hooks.c b/security/selinux/hooks.c
index d7539de..37ae497 100644
--- a/security/selinux/hooks.c
+++ b/security/selinux/hooks.c
@@ -1338,6 +1338,8 @@ static inline u16 socket_type_to_security_class(int family, int type, int protoc
 		return SECCLASS_KEY_SOCKET;
 	case PF_APPLETALK:
 		return SECCLASS_APPLETALK_SOCKET;
+	case PF_CAN:
+		return SECCLASS_CAN_SOCKET;
 	}
 
 	return SECCLASS_SOCKET;
diff --git a/security/selinux/include/classmap.h b/security/selinux/include/classmap.h
index 1f1f4b2..a67d13c 100644
--- a/security/selinux/include/classmap.h
+++ b/security/selinux/include/classmap.h
@@ -165,5 +165,7 @@ struct security_class_mapping secclass_map[] = {
 	  { COMMON_CAP_PERMS, NULL } },
 	{ "cap2_userns",
 	  { COMMON_CAP2_PERMS, NULL } },
+	{ "can_socket",
+	  { COMMON_SOCK_PERMS, NULL } },
 	{ NULL }
   };
-- 
1.7.5.4

